src/sage/combinat/posets/hochschild_lattice.py

```diff
@@ -0,0 +1,147 @@
+
+The underlying set of `H(n)` consists of some words in the alphabet `(0,1,2)`, whose precise description can be found in [Com2021]_.
+"""
+from typing import Iterator
+
+from sage.combinat.posets.lattices import LatticePoset
+from sage.misc.lazy_import import lazy_import
+from sage.modules.free_module_element import vector
+from sage.topology.simplicial_complex import SimplicialComplex
+
+lazy_import("sage.geometry.cone", "Cone")
+lazy_import("sage.geometry.fan", "Fan")
+
+
+def hochschild_lattice(n) -> LatticePoset:
+    r"""
+    Return the Hochschild lattice `H(n)`.
+
+    INPUT:
+
+    - `n \geq 1` -- an integer
+
+    The cardinality of `H(n)` is `2^{n - 2} \times (n + 3)`.
+
+    EXAMPLES::
+
+        sage: P = posets.HochschildLattice(5); P
+        Finite lattice containing 64 elements
+        sage: P.degree_polynomial()
+        x^5 + 9*x^4*y + 22*x^3*y^2 + 22*x^2*y^3 + 9*x*y^4 + y^5
+
+    TESTS::
+
+        sage: posets.HochschildLattice(0)
+        Traceback (most recent call last):
+        ...
+        ValueError: this requires n >= 1
+    """
+    if n <= 0:
+        raise ValueError("this requires n >= 1")
+
+    def iterator_True(n) -> Iterator[tuple[int, ...]]:
+        """
+        Iterator over part of the underlying set.
+        """
+        if n == 0:
+            yield ()
+            return
+        for w in iterator_True(n - 1):
+            yield (0,) + w
+            yield (2,) + w
+
+    def iterator_False(n) -> Iterator[tuple[int, ...]]:
+        """
+        Iterator over rest of the underlying set.
+        """
+        if n == 0:
+            yield ()
+            return
+        for w in iterator_True(n - 1):
+            yield (0,) + w
+        for w in iterator_False(n - 1):
+            yield (1,) + w
+            yield (2,) + w
+
+    def sommets(n) -> Iterator[tuple[int, ...]]:
+        """
+        Iterator over the full underlying set.
+        """
+        for w in iterator_True(n - 1):
+            yield (0,) + w
+        for w in iterator_False(n - 1):
+            yield (1,) + w
+
+    verts = list(sommets(n))
+
+    def compare(a, b) -> bool:
+        return all(ai <= bi for ai, bi in zip(a, b))
+
+    return LatticePoset([verts, compare])
+
+
+def hochschild_fan(n):
+    """
+    Return Saneblidze's fan for the Hochschild polytope.
+
+    The dual polytope is obtained from a standard simplex
+    by a sequence of truncations.
+
+    EXAMPLES::
+
+        sage: from sage.combinat.posets.hochschild_lattice import hochschild_fan
+        sage: F = hochschild_fan(4); F
+        Rational polyhedral fan in 4-d lattice N
+        sage: F.f_vector()
+        (1, 11, 39, 56, 28)
+    """
+    rays = [vector([1] * n)]
+    rays.extend(vector([0 if j != i else -1 for j in range(n)])
+                for i in range(n))
+
+    cones = [Cone([rays[j] for j in range(n + 1) if j != i])
+             for i in range(n + 1)]
+
+    # standard fan of projective space Pn
+    F = Fan(cones)
```

Comment:
```suggestion
    """
    from sage.geometry.cone import Cone
    from sage.geometry.fan import Fan
    from sage.modules.free_module_element import vector

    rays = [vector([1] * n)]
    rays.extend(vector([0 if j != i else -1 for j in range(n)])
                for i in range(n))

    cones = [Cone([rays[j] for j in range(n + 1) if j != i])
             for i in range(n + 1)]

    # standard fan of projective space Pn
    F = Fan(cones, check=False)
```
More local imports and disabling the unnecessary check in `Fan` to speed up the construction.